docs/testplan/console/standalone_sonic_console_server_test_plan.md

```diff
@@ -0,0 +1,356 @@
+
+![c0 topo](c0_topo_s3.drawio.svg)
+
+#### Scenario #4 Provision data path (for M1)
+
+This is only scenario that the C0 is not the endpoint in the network. It is been used as a L3 ethernet switch in this scenario and forward traffic from `Secondary Network` to devices that not reachable from `Primary Network` directly.
+
+![c0 topo](c0_topo_s4.drawio.svg)
+
+### Physical Testbed
+
+#### Pinouts defination
+
+Below table illustrated the console server DUT's DCE ports' pinouts 
+
+| PIN number | Signal |   |
+| ---------- | ------ | - |
+| 1 | CTS | Clear To Send |
+| 2 | DSR | Data Set Read |
+| 3 | RXD |  |
+| 4 | GND |  |
+| 5 | GND |  |
+| 6 | TXD |  |
+| 7 | DTR | Data Terminal Ready |
+| 8 | RTS | Request To Send |
+
+#### Testbed Setup - c0
+
+Below diagram shows the physical testbed setup:
+
+![c0 topo](c0_topo_phy.drawio.svg)
+
+To build a c0 topo testbed, we need prepare 2 kinds of leaf fanout:
+
+- Regular ethernet switch for ethernet ports on the DUT
+
+- Another console server for all console ports 
+
+Unlike regular leaf fanout switch, they have ethernet connection with root fanout and then server, the console server leaf fanout does not need ethernet connection with server via the root fanout as it does not have direct ethernet connection with DUT. It will have indirect connectvitity with server/vms via the management plane.
+
+The cable used between DUT and console serer Leaf Fanout are rollover cables. The pinouts mapping can refer to below table:
+
+| Color | DUT PIN number | DUT Signal | L-Fan Signal | L-Fan PIN number |
+| - | - | - | - | - |
+| White-Orange | 1 | CTS | RTS | 8 |
+| Orange | 2 | DSR | DTR | 7 |
+| White-Green | 3 | RXD | TXD | 6 | 
+| Green | 4 | GND | GND | 5 |
+| White-Blue | 5 | GND | GND | 4 | 
+| Blue | 6 | TXD | RXD | 3 |
+| White-Brown | 7 | DTR | DSR | 2 | 
+| Brown | 8 | RTS | CTS  | 1 |
+
+##### Limitation
+
+- The console serve leaf fanout cannot pass through the hardware flowcontrol signals which result in the test can be only proceed for normal data transfer related test cases
+
+#### Testbed Setup - c0-lo
+
+Below diagram shows the phsycial testbed setup for c0-lo:
+
+![c0 topo](c0_lo_topo_phy.drawio.svg)
+
+To build a c0-lo topo testbed, we need prepare just ethernet leaf fanout. All the console ports on DUT will be plugged with RJ45 loopback module:
+
+![c0 topo](img/loopback_module.png)
+
+The pins out and connection are illusrated as below:
+
+| PIN number | connect to PIN number |
+| - | - |
+| 1 | 8 |
+| 2 | 7 |
+| 3 | 6 |
+| 4 | 5 |
+
+* Depends on the hardware flow control implementaion, we might able to ignore the 2-7 connection 
+
+## Test Cases
+
+Dataplane test will be covered by existing tests. 
+
+Following features are mandatory:
+- arp
+- autorestart
+- bgp
+- bmp
+- cacl
+- container_checker
+- database
+- disk
+- dns
+- dut_console
+- fips
+- generic_config_updater
+- gnmi
+- golden_config_infra
+- iface_namingmode
+- ip
+- ipfwd
+- kubesonic
```

Comment:
Do we really have usecase to run Kubernetes for console/terminal server?

Reply:
We use k8s to upgrade management related containers/non-dataplane impact containers such as telemetry/gnmi/...